src/commands/fun/higherlower_game/higherlower.rs

```diff
@@ -0,0 +1,627 @@
+            "You can't play two higher lower games at once! \n\
+            Finish your [other game](https://discord.com/channels/{}/{}/{}) first or give up.",
+            match game.guild {
+                Some(id) => id.to_string(),
+                None => "@me".to_string(),
+            },
+            game.channel,
+            game.id
+        )
+    });
+
+    if let Some(content) = content {
+        let components = give_up_components();
+        let embed = EmbedBuilder::new().color(RED).description(content).build();
+
+        let builder = MessageBuilder::new().embed(embed).components(&components);
+        command.create_message(&ctx, builder).await?;
+
+        Ok(())
+    } else {
+        let (play1, mut play2) =
+            match tokio::try_join!(random_play(&ctx, 0.0, 0), random_play(&ctx, 0.0, 0)) {
+                Ok(tuple) => tuple,
+                Err(err) => {
+                    let _ = command.error(&ctx, GENERAL_ISSUE).await;
+                    return Err(err);
+                }
+            };
+        while play2 == play1 {
+            play2 = random_play(&ctx, 0.0, 0).await?;
+        }
+
+        //TODO: handle mode
+        let mut game = HlGameState {
+            previous: play1,
+            next: play2,
+            player: user,
+            id: Id::new(1),
+            channel: command.channel_id(),
+            guild: command.guild_id(),
+            mode: 1,
+            current_score: 0,
+            highscore: ctx.psql().get_higherlower_highscore(user.get(), 1).await?,
+        };
+
+        let image = game.create_image(&ctx).await?;
+        let components = hl_components();
+        let embed = game.to_embed(image);
+
+        let builder = MessageBuilder::new().embed(embed).components(&components);
+        let response = command.create_message(&ctx, builder).await?.model().await?;
+        game.id = response.id;
+        ctx.hl_games().insert(user, game);
+
+        Ok(())
+    }
+}
+
+async fn random_play(ctx: &Context, prev_pp: f32, curr_score: u32) -> BotResult<HlGameStateInfo> {
+    let max_play = 1.max(25 - curr_score);
+    let (rank, play): (u32, u32) = {
+        let mut rng = rand::thread_rng();
+        (rng.gen_range(1..=5000), rng.gen_range(0..max_play))
+    };
+
+    let page = ((rank - 1) / 50) + 1;
+    let idx = (rank - 1) % 50;
+
+    // ! Currently 3 requests, can probably be reduced
+    let player = ctx
+        .osu()
+        .performance_rankings(GameMode::STD)
+        .page(page)
+        .await?
+        .ranking
+        .swap_remove(idx as usize);
+
+    let mut plays = ctx
+        .osu()
+        .user_scores(player.user_id)
+        .limit(50)
+        // .offset(play as usize)
+        .mode(GameMode::STD)
+        .best()
+        .await?;
+
+    plays.sort_by(|a, b| {
+        (round(a.pp.unwrap_or(0.0)) - prev_pp)
+            .abs()
+            .partial_cmp(&(round(b.pp.unwrap_or(0.0)) - prev_pp).abs())
+            .unwrap()
+    });
```

Comment:
`sort_unstable_by` should be sufficient and more efficient. And `round` is probably not important here.

Reply:
👍